apps/wallet-mobile/src/Staking/PoolTransition/usePoolTransition.tsx

```diff
@@ -0,0 +1,426 @@
+  return new Promise((resolve) => {
+    if (currentPoolId === 'd248ded3c18e0e80d07a46f00a2d808075b989ccb1a0e40a76e5cee1') {
+      resolve({
+        currentPool: {
+          pic: 'https://img.cexplorer.io/a/5/5/f/2/pool12eht6dqxpzqj87xuextrpufz2gxmt4reuesuw26r2utzw0kw906.png',
+          ticker: 'EMRG',
+          name: 'emurgo old',
+          roa: '0',
+          tax: '3.5',
+        } as unknown as PoolInfo & {tax: number},
+        suggestedPool: {
+          pic: 'https://img.cexplorer.io/e/c/2/3/1/pool1dkww2vlysa8lsnuf5rca979zdsyr3zvt59hu7e420yxfunkka2z.png',
+          id: 'df1750df9b2df285fcfb50f4740657a18ee3af42727d410c37b86207',
+          name: 'emurgo new',
+          roa: '5.1',
+          ticker: 'EMRG',
+          tax: '3.2',
+        } as unknown as PoolInfo & {tax: number},
+        deadlineMilliseconds: 2999777000,
+      })
+    } else {
+      resolve(null)
+    }
+  })
+}
+
+export const usePoolTransition = () => {
+  const navigation = useNavigation()
+  const wallet = useSelectedWallet()
+  const {stakingInfo} = useStakingInfo(wallet)
+
+  const isStaked = stakingInfo?.status === 'staked'
+  const currentPoolId = isStaked ? stakingInfo?.poolId : ''
+
+  const poolTransitionQuery = useQuery({
+    enabled: isStaked,
+    retry: false,
+    staleTime: 5 * 60 * 1000,
+    queryKey: [wallet.id, 'poolTransition', currentPoolId],
+    queryFn: () => getPoolTransition(currentPoolId),
+  })
+
+  const poolTransition = poolTransitionQuery.data ?? null
+  const poolId = poolTransition?.suggestedPool.id ?? ''
+
+  const stakingTxQuery = useQuery({
+    enabled: poolId !== '',
+    retry: false,
+    staleTime: 5 * 60 * 1000,
+    queryKey: [wallet.id, 'stakingTx', poolId],
+    queryFn: async () => {
+      const accountStates = await wallet.fetchAccountState()
+      const accountState = accountStates[wallet.rewardAddressHex]
+      // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
+      if (!accountState) throw new Error('Account state not found')
+
+      const stakingUtxos = await wallet.getAllUtxosForKey()
+      const amountToDelegate = Quantities.sum([
+        ...stakingUtxos.map((utxo) => utxo.amount as Balance.Quantity),
+        accountState.remainingAmount as Balance.Quantity,
+      ])
+
+      return wallet.createDelegationTx(poolId, new BigNumber(amountToDelegate))
+    },
+  })
+
+  const yoroiUnsignedTx = stakingTxQuery.data
+
+  const navigateToUpdate = () =>
+    !yoroiUnsignedTx
+      ? null
+      : navigation.navigate('app-root', {
+          screen: 'main-wallet-routes',
+          params: {
+            screen: 'staking-dashboard',
+            params: {
+              screen: 'delegation-confirmation',
+              initial: false,
+              params: {
+                poolId,
+                yoroiUnsignedTx,
+              },
+            },
+          },
+        })
+
+  return {
+    poolTransition,
+    canTransitionPool: !!yoroiUnsignedTx,
```

Comment:
did Product provide a "not enough balance" to create the TX, or the state for that is just, "not showing the modal" or "disabling the re-delegation" ?

Reply:
I asked for their input, and a tentative new screen for that was suggested: https://www.figma.com/file/KeBVo6rWNYFLXwTphz9lHi/Mobile-App?type=design&node-id=20568-39794&mode=design&t=P59sb3i9dFWiSh4I-0

I suggested the course of not showing the modal if the tx failed to be created, until the portfolio changes affected the staking center.

Reply:
Following that in slack with them